src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs

```diff
@@ -298,16 +296,14 @@ public ColorBlend InterpolationColors
             }
             set
             {
-                // The Desktop implementation will throw various exceptions - ranging from NullReferenceExceptions to Argument(OutOfRange)Exceptions
-                // depending on how sane the input is. These checks exist to replicate the exact Desktop behavior.
                 int count = value.Colors.Length;
 
                 if (value.Positions == null)
-                    throw new ArgumentNullException("source");
+                    throw new ArgumentException(SR.Format(SR.InvalidArgumentValue, "value.Positions", value.Positions), nameof(value));
                 if (value.Colors.Length > value.Positions.Length)
-                    throw new ArgumentOutOfRangeException();
+                    throw new ArgumentOutOfRangeException(nameof(value));
                 if (value.Colors.Length < value.Positions.Length)
```

Comment:
If the &gt; and &lt; are now throwing the same exception, perhaps this should just be folded into one !=.

Reply:
yep, will update